patches/Memcache.patch

```diff
@@ -0,0 +1,11 @@
+--- memcache/memcache.php	2023-09-04 17:18:27.000000000 +0200
++++ memcache/memcache.php	2023-09-08 14:29:48.807174822 +0200
+@@ -217,7 +217,7 @@
+      * The lowest byte of the int is reserved for pecl/memcache internal usage (e.g. to indicate
+      * compression and serialization status).
+      * </p>
+-     * @return string|array|false <p>
```

Comment:
This broke `memcache-get.php` test. The explanation is here: https://github.com/phpstan/phpstan-src/pull/2323#issuecomment-1497231186

Reply:
This is not (easily) possible to fix via `functionMap.php`. There are two variants in function map: one which returns `mixed` and one which returns `array|false`. And as you can see, the phpdoc return type in phpstorm-stubs is `string|array|false`.

`decideType(mixed, string|array|false) = string|array|false`, which doesn't seem like something that could be changed.

Perhaps if `decideType` also considered where the two types come from. E.g. in this case we know that the first type comes from the function map. Since `string|array|false` could have been used in the function map, we could reach the conclusion that we used the broader type intentionally. However, there are issues in the function map as well, so it's though to say whether such approach would be more feasible.

Reply:
What about disregarding PhpStorm stubs type completely if functionMap has multiple variants? But the "golden test" for Reflection will come in handy in that scenario :)

Reply:
Nice, that worked (for memcace::get). I limited it specifically to the return type. The stubs are still useful/necessary for positional parameter names, `@throws`, ...

But unfortunately, it doesn't cover the `ReflectionPropety` patch.